sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/CastOperatorConversion.java

```diff
@@ -76,6 +76,8 @@ public class CastOperatorConversion implements SqlOperatorConversion
       builder.put(type, ExprType.LONG);
     }
 
+    builder.put(SqlTypeName.ARRAY, ExprType.ARRAY);
```

Comment:
this isn't probably that cool because it loses the array element type. I think maybe #13890 has a better solution for this by switching cast to use `Calcites.getColumnTypeForRelDataType`

Reply:
Yeah, yours is better. Merged it in. Thanks.